doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -284,7 +284,7 @@ There is dedicated syntax for list and array literals.
    tac2rec_fieldpats ::= @tac2rec_fieldpat ; {? @tac2rec_fieldpats }
    | @tac2rec_fieldpat ;
    | @tac2rec_fieldpat
-   tac2rec_fieldpat ::= @qualid := @tac2pat1
+   tac2rec_fieldpat ::= @qualid {? := @tac2pat1 }
```

Comment:
How about adding a sentence or 2 below describing the punning behavior?  (All of these items should eventually be explained.  Best to capture that now even if it's the only item described.)

Reply:
What about
> If a field of an Ltac2 record has the name of a variable in scope, then the assignment `:=` does not need to be written, allowing one to shorten `myvar := myvar;` to `myvar;`. Or something like that.

Reply:
Another question that I've thought about is if punning makes sense as a feature for Coq records as well?

Reply:
added a bit